pkg/virt-controller/watch/vm.go

```diff
@@ -2740,6 +2766,15 @@ func (c *VMController) sync(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachin
 		return vm, nil, nil
 	}
 
+	// Using the empty printableStatus as an indication that the VM was just created.
+	if vm.Status.PrintableStatus == "" && vm.Spec.RunStrategy != nil && *vm.Spec.RunStrategy == virtv1.RunStrategyRerunOnFailure {
```

Comment:
Sorry, I haven't looked through the whole PR, but the comment here raised my attention. There is currently another PR https://github.com/kubevirt/kubevirt/pull/10846 which sets a default value to `vm.Status.PrintableStatus`. Won't it break the logic here? Ping @RamLavi

Reply:
Nice catch @vasiliy-ul !! I totally does break the logic... If the same `kubebuilder` syntax can be used conditionally on a spec field then I can properly fix this. If not I'll have to find another way.

Reply:
AFAIK there is no way to use the kubebuilber conditionally :(

Reply:
I think you're right @fossedihelm ...
Any idea on how to get the info "VM just got created"?
I could add a new `StateChangeRequestAction` like "StartIfRequired" and add it to the `StateChangeRequests` of a new VM on creation with kubebuilder.
I could also create a whole new status field, like a "JustCreated" / "NeverStarted" / "EverStarted" boolean.
We could also set PrintableStatus to something like "New" instead of "Stopped" by default, that would also give more info to users but might be confusing...
I'll suggest the latter on Ram's PR.

Reply:
TBH, `PrintableStatus` does not look like smth you can securely rely on to detect that `VM just got created`. The name `Printable...` suggests that it is more like a field for pretty printing in the CLI output or for showing it in UI. I am not sure that introducing a new value ("New" instead of "Stopped" by default) is a robust solution, as this printable status may be accidentally changed in the future, thus again breaking the logic here. But it is my personal perception. I might be wrong here and just got confused by the naming (Printable...).

Reply:
I have not reviewed the PR/comments but please do not rely on `PrintableStatus` as this is the best-effort field meant for user interfaces only. Nor external logic or internal logic should rely on it.

Reply:
Instead of using the kubebuilder, how about to use a `StartRequest` (in the create mutator webhook?) in case `RerunOnFailure` is used?

Reply:
That was the first thing I tried, but it doesn't work. I think mutating webhooks are not allowed to alter status fields.

Reply:
Right! Now I remember... :man_facepalming:

Reply:
Switched to a condition. PTAL!